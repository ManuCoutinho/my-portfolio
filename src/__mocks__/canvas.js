HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
