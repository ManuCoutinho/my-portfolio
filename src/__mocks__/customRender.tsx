import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
