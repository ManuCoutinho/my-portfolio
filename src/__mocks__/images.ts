export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
