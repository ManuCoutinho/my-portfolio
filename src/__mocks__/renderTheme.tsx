import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
