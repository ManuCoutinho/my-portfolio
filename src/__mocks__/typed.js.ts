/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

export default class Typed {
  constructor(elementId: string, options: any) {}
  toggle(): void {}
  stop(): void {}
  start(): void {}
  destroy(): void {}
  reset(restart?: boolean): void {}
}
