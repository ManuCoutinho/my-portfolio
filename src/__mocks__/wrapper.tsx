/* eslint-disable react/display-name */
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { DataProviderMock } from '__mocks__/providers/DataContext'

export const customWrapper = () => {
	return ({ children }: { children: React.ReactNode }) => (
		<ThemeContextProvider>
			<DataProviderMock>{children}</DataProviderMock>
		</ThemeContextProvider>
	)
}
