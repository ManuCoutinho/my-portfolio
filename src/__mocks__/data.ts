import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
