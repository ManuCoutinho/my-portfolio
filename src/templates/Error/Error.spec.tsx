import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import { Error } from '.'

describe('<Error />', () => {
	it('should render the Erro template correctly', () => {
		const { container } = render(<Error />)
		expect(
			screen.getByRole('heading', { name: /Opss! Como você chegou aqui?/ })
		).toBeInTheDocument()
		expect(
			screen.getByText('Parece que você chegou o limite deste universo.')
		).toBeInTheDocument()
		expect(screen.getByAltText('error image')).toBeInTheDocument()
		expect(screen.getByRole('button', { name: /voltar/i })).toBeInTheDocument()
		expect(container).toMatchSnapshot()
	})
})
