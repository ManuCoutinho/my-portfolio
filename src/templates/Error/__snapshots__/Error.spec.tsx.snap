// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render the Erro template correctly 1`] = `
<div>
  <div
    class="sc-ce9dac95-0 QWJlT"
    data-testid="stackbox"
  >
    <div
      align="center"
      class="sc-ce9dac95-0 khteWf"
      data-testid="stackbox"
      direction="column"
      gap="1"
    >
      <div
        class="sc-ce9dac95-0 dEZKmI"
        data-testid="stackbox"
        justify="center"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27800%27%20height=%27500%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="error image"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        align="center"
        class="sc-ce9dac95-0 jrMRLA"
        data-testid="stackbox"
        direction="column"
        gap="1"
        justify="center"
      >
        <h3
          class="sc-94e9b77b-0 dbunNU"
        >
          Opss! Como você chegou aqui?
        </h3>
        <p
          align="center"
          class="sc-f89260d2-0 cKMxeN"
        >
          Parece que você chegou o limite deste universo.
        </p>
        <button
          class="sc-66462b9f-0 hLCyMx"
          type="button"
          variant="ghost"
        >
          Voltar
        </button>
      </div>
    </div>
  </div>
</div>
`;
